aws-analytics-pinpoint/src/main/java/com/amplifyframework/analytics/pinpoint/Session.kt

```diff
@@ -64,25 +67,22 @@ internal class Session {
         }
     }
 
-    private fun generateSessionId(): String {
+    private fun generateSessionId(uniqueId: String): String {
         val sessionIdTimeFormat = SimpleDateFormat("yyyyMMdd-HHmmssSSS", Locale.US)
         sessionIdTimeFormat.timeZone = TimeZone.getTimeZone("UTC")
-        val uniqueId = "" // TODO(": Get uniqueId from shared preferences.")
         val time = sessionIdTimeFormat.format(startTime)
-        return "${trimOrPad(uniqueId, maxSessionIdLength, sessionIdPaddingChar)}$sessionIdDelimiter$time"
+        return "${trimOrPad(uniqueId)}$sessionIdDelimiter$time"
     }
 
     private fun trimOrPad(
-        input: String,
-        outputLength: Int,
-        paddingCharacter: String
+        input: String
     ): String {
         val stringBuffer = StringBuffer()
-        if (input.length > outputLength - 1) {
-            stringBuffer.append(input.substring(input.length - outputLength))
+        if (input.length > maxSessionIdLength - 1) {
+            stringBuffer.append(input.substring(input.length - maxSessionIdLength))
```

Comment:
Why is the input trimmed to length of `input.length - maxSessionIdLength` rather than `maxSessionIdLength`?

Reply:
Legacy sdk trims the last x chars incase the length > maxLength

https://github.com/aws-amplify/aws-sdk-android/blob/67a111dfaee587ddb9c319a25e77310ab8cd706e/aws-android-sdk-pinpoint/src/main/java/com/amazonaws/mobileconnectors/pinpoint/internal/core/util/StringUtil.java#L148